be/src/exec/orc_builder.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+    auto type = iter->second;
+    switch (starrocks_type.type) {
+    case TYPE_CHAR: {
+        if (len < 0) {
+            len = starrocks::TypeDescriptor::MAX_CHAR_LENGTH;
+        }
+        result = orc::createCharType(orc::CHAR, len);
+        break;
+    }
+    case TYPE_VARCHAR: {
+        if (len < 0) {
+            len = starrocks::TypeDescriptor::MAX_VARCHAR_LENGTH;
+        }
+        result = orc::createCharType(orc::VARCHAR, len);
+        break;
+    }
+    case TYPE_DECIMAL128:
+    case TYPE_DECIMAL64:
+    case TYPE_DECIMAL32:
+    case TYPE_DECIMALV2: {
+        result = orc::createDecimalType(precision, scale);
+        break;
+    }
+    default: {
+        result = orc::createPrimitiveType(type);
+    }
+    }
+
+    return Status::OK();
+}
+
+const size_t ORCBuilder::OUTSTREAM_BUFFER_SIZE_BYTES = 1024 * 1024;
+
+ORCBuilder::ORCBuilder(ORCBuilderOptions options, std::unique_ptr<WritableFile> writable_file,
+                       const std::vector<ExprContext*>& output_expr_ctxs,
+                       TupleDescriptor* _output_tuple_desc,
+                       std::vector<std::string>  column_names,
+                       std::map<std::string, TIcebergTypeAttribute> attributes)
+        : _options(options),
+          _output_expr_ctxs(output_expr_ctxs),
+          _output_tuple_desc(_output_tuple_desc),
+          _batchSize(1024),
+          _init(false),
+          _writable_file(std::move(writable_file)),
+          _column_names(std::move(column_names)),
+          _attributes(std::move(attributes)) {}
+
+Status ORCBuilder::init(vectorized::Chunk* chunk) {
+    if (_init || chunk->num_rows() == 0) {
+        return Status::OK();
+    }
+
+    size_t num_fields = _output_expr_ctxs.size();
+    std::unique_ptr<orc::Type> schema = orc::createStructType();
+
+    for (int idx = 0; idx < num_fields; idx++) {
+        // resolve column type
+        const auto type = _output_expr_ctxs.at(idx)->root()->type();
+        std::unique_ptr<orc::Type> fieldType;
+        Status status = _starrocks_type_to_orc_type_descriptor(type, fieldType);
+        if (!status.ok()) {
+            return status;
+        }
+
+        // resolve column name
+        auto root = _output_expr_ctxs.at(idx)->root();
+        auto column_id = root->get_column_ref()->slot_id();
+        string column_name;
+        if (chunk->schema() != nullptr && !chunk->get_column_name(column_id).empty()) {
+            column_name = chunk->get_column_name(column_id);
+        } else if (_output_tuple_desc != nullptr) {
+            column_name = _output_tuple_desc->slots()[idx]->col_name();
+        } else if (_column_names.size() > idx) {
+            column_name = _column_names[idx];
+        } else {
+            column_name = "col" + std::to_string(idx);
+        }
+
+        // fill iceberg type attribute
+        auto it = this->_attributes.find(column_name);
+        if (it != this->_attributes.end()) {
```

Comment:
It's very wierd to appear `iceberg` in `orc_builder` class, looks like they can be decoupled right? Could you make the `orc_builder` class more general, instead of tightly coupling with iceberg? (I understand your intention is to write in iceberg format).